internal/services/appservice/linux_function_app_resource.go

```diff
@@ -523,9 +533,14 @@ func (r LinuxFunctionAppResource) Create() sdk.ResourceFunc {
 					ClientCertMode:       pointer.To(webapps.ClientCertMode(functionApp.ClientCertMode)),
 					DailyMemoryTimeQuota: pointer.To(functionApp.DailyMemoryTimeQuota), // TODO - Investigate, setting appears silently ignored on Linux Function Apps?
 					VnetRouteAllEnabled:  siteConfig.VnetRouteAllEnabled,
+					VnetImagePullEnabled: pointer.To(functionApp.VnetImagePullEnabled),
 				},
 			}
 
+			if isAseEnvironment && !functionApp.VnetImagePullEnabled {
+				return fmt.Errorf("`vnet_image_pull_enabled` cannot be disabled for app running in an app service environment.")
+			}
+
```

Comment:
Let's move this into  the`if` block on line 415 and simplify it to 
```suggestion
			if !functionApp.VnetImagePullEnabled {
				return fmt.Errorf("`vnet_image_pull_enabled` cannot be disabled for app running in an app service environment.")
			}

```